src/mds/SessionMap.cc

```diff
@@ -924,6 +983,35 @@ int Session::check_access(CInode *in, unsigned mask,
   return 0;
 }
 
+void SessionMap::hit_session(Session *session) {
+  auto it = session_load_avg_map.find(session->info.inst.name);
+  assert(it != session_load_avg_map.end());
+
+  auto &session_load_avg = it->second;
+  session_load_avg.adjust();
+
+  // recalculate total load and average across sessions
+  double total_load = 0.0;
+  double avg_load;
+
+  int sessions = 0;
+  for (auto const &it : session_load_avg_map) {
```

Comment:
nit: `it` would indicate an iterator variable but it is actually a `std::pair`. Suggest using `p` instead.

Reply:
ok -- habit of interpreting it as *iterating* over a map :)